apps/gateway/lib/gateway/controllers/arena/currency_controller.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Gateway.Controllers.Users.Currency.CurrencyController do
+  use Gateway, :controller
+
+  alias GameBackend.Users.Currencies.UserCurrency
+  alias GameBackend.Users.Currencies
+
+  @moduledoc """
+    Controller to control currency changes in users
+  """
+
+  def modify_currency(conn, %{"currency_name" => currencty_name, "amount" => amount, "user_id" => user_id}) do
+    with {:value, {amount, _}} <- {:value, Integer.parse(amount)},
```

Comment:
This should not be needed, the JSON decoder in `:api` will take care of this

Reply:
deleted!